src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+        block.rktSolver = ReaktoroSolver(
+            block.rkt_state,
+            block.rkt_inputs,
+            block.rkt_outputs,
+            block.rkt_jacobian,
+            block_name=name,
+        )
+        if speciation_block:
+            presolve = self.config.presolve_speciation_block
+        else:
+            presolve = self.config.presolve_property_block
+        block.rktSolver.set_solver_options(
+            tolerance=self.config.tolerance,
+            epsilon=self.config.epsilon,
+            presolve=presolve,
+            presolve_tolerance=self.config.presolve_tolerance,
+            presolve_epsilon=self.config.presolve_epsilon,
+            max_iters=self.config.max_iters,
+            presolve_max_iters=self.config.presolve_max_iters,
+            hessian_type=self.config.hessian_type,
+        )
+
+    def build_gray_box(self, block):
+        """this will build rkt outputs specified block.
+        The keyword arguments are for automatic configuration of speciation and property blocks
+
+        Keywords:
+        block -- pyomo block to build the model on
+        """
+        """ build block"""
+        scaling = self.config.jacobian_user_scaling
+        scaling_type = self.config.jacobian_scaling_type
+        block.rkt_block_builder = ReaktoroBlockBuilder(
+            block, block.rktSolver, build_on_init=False
+        )
+        block.rkt_block_builder.configure_jacobian_scaling(
+            jacobian_scaling_type=scaling_type, user_scaling=scaling
+        )
+        block.rkt_block_builder.build_reaktoro_block()
+
+    def display_jacobian_outputs(self):
+        if self.config.build_speciation_block:
+            _log.info("-----Displaying information for speciation block ------")
+            self.speciation_block.rkt_jacobian.display_jacobian_output_types()
+        _log.info("-----Displaying information for property block ------")
+        self.rkt_jacobian.display_jacobian_output_types()
+
+    def display_jacobian_scaling(self):
+        jacobian_scaling = {}
+        if self.config.build_speciation_block:
+            _log.info("-----Displaying information for speciation block ------")
+            jac_scale = (
+                self.speciation_block.rkt_block_builder.display_jacobian_scaling()
+            )
+            jacobian_scaling["speciation_block"] = jac_scale
+        _log.info("-----Displaying information for property block ------")
+        jac_scale = self.rkt_block_builder.display_jacobian_scaling()
+        jacobian_scaling["property_block"] = jac_scale
+        return jacobian_scaling
+
+    def set_jacobian_scaling(self, user_scaling_dict, speciation_block=False):
+        if speciation_block:
+            self.speciation_block.rkt_block_builder.set_user_jacobian_scaling(
+                user_scaling_dict
+            )
+        else:
+            self.rkt_block_builder.set_user_jacobian_scaling(user_scaling_dict)
+
+    def initialize(self):
```

Comment:
Is this equivalent to the existing `process_block.initialize()` method from IDAES?

If so, we should probably instead move to an [`Initializer` object](https://idaes-pse.readthedocs.io/en/stable/reference_guides/initialization/developing_initializers.html?highlight=Initializer), as we plan to deprecate the `model.initialize()` API in the future. For backward compatibility, we can keep the `initialize` method, but have it just wrap the `Initializer` instead.

Reply:
I will put a todo for this.